numpyro/contrib/reparam.py

```diff
@@ -97,10 +97,16 @@ class TransformReparam(Reparam):
     """
     def __call__(self, name, fn, obs):
         assert obs is None, "TransformReparam does not support observe statements"
+        batch_shape = fn.batch_shape
+        if isinstance(fn, dist.ExpandedDistribution):
+            fn = fn.base_dist
```

Comment:
We have `_wrap/_unwrap` logic in pyro's `Reparam` for [Independent](https://github.com/pyro-ppl/pyro/blob/73af4ff/pyro/infer/reparam/reparam.py). Do you think having something like that for expanded and masked distributions might be helpful here?

Reply:
I think something like `_unexpand` and `_expand` would do the job here. Let's make those helpers when we have more reparameterization classes and the logic here is repeated. I am not sure about Masked because I would like to avoid dealing with jacobian in TransformReparam.